assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -275,43 +276,114 @@ export const selectors = {
 	 */
 	haveConversionEventsForTailoredMetrics: createRegistrySelector(
 		( select ) => ( state, useNewEvents ) => {
-			const leadRelatedMetrics = [
-				KM_ANALYTICS_TOP_PAGES_DRIVING_LEADS,
-				KM_ANALYTICS_TOP_CITIES_DRIVING_LEADS,
-				KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_LEADS,
-			];
-			const conversionEventWidgets = {
-				purchase: [
-					KM_ANALYTICS_TOP_CITIES_DRIVING_PURCHASES,
-					KM_ANALYTICS_TOP_DEVICE_DRIVING_PURCHASES,
-					KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_PURCHASES,
-				],
-				add_to_cart: [
-					KM_ANALYTICS_TOP_CITIES_DRIVING_ADD_TO_CART,
-					KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_ADD_TO_CART,
-				],
-				contact: leadRelatedMetrics,
-				submit_lead_form: leadRelatedMetrics,
-				generate_lead: leadRelatedMetrics,
-			};
-
-			const purposeTailoredMetrics = select(
-				CORE_USER
-			).getAnswerBasedMetrics( null, true );
-
 			const conversionReportingEventsChange = useNewEvents
 				? select(
 						MODULES_ANALYTICS_4
 				  ).getConversionReportingEventsChange()?.newEvents
 				: select( MODULES_ANALYTICS_4 ).getDetectedEvents();
 
-			return conversionReportingEventsChange?.some( ( event ) =>
+			const currentTailoredMetrics =
+				select( CORE_USER ).getAnswerBasedMetrics();
+
+			const tailoredMetricsWithNewEvents = select(
+				CORE_USER
+			).getAnswerBasedMetrics( null, conversionReportingEventsChange );
+
+			return tailoredMetricsWithNewEvents?.some(
+				( metric, index ) =>
+					metric !== currentTailoredMetrics?.[ index ]
+			);
```

Comment:
This has the potential to return `undefined`, so the documentation should be updated.